docs/fsharp-cheatsheet.md

```diff
@@ -169,197 +205,281 @@ The `rec` keyword is used together with the `let` keyword to define a recursive
+        [ 1
+          2 ]
+    let list3 = "c" :: list1   // prepending; [ "c";  "a"; "b" ]
+    let list4 = list1 @ list3  // concat; [ "a"; "b";  "c"; "a"; "b" ]
+    let list5 = [ 1..2..9 ]    // start..skip..last; [ 1; 3; 5; 7; 9 ]
+
+    // Slicing is inclusive
+    let firstTwo = list5[0..1]  // [ 1; 3 ]
+
+    // Pattern matching
+    match myList with
+    | [] -> ...            // empty list
+    | [ 3 ] -> ...         // single 3 item
+    | [ _; 4 ] -> ...      // use of wild card
+    | head :: tail -> ...  // cons pattern. head is the first item, tail is the rest
+
+    // Recursive calls with a list using cons pattern
+    let sumEachItem (myList:int list) =
+        match myList with
         | [] -> 0
-        | x :: xs -> x + sum xs
+        | head :: tail -> head + sumEachItem tail
+
+See the [List Module](https://fsharp.github.io/fsharp-core-docs/reference/fsharp-collections-listmodule.html) for built-in functions.
+
+<div id="collections-arrays"></div>
 
 ### Arrays
+
 *Arrays* are fixed-size, zero-based, mutable collections of consecutive data elements.
 
-    // Arrays use square brackets with bar
-    let array1 = [| "a"; "b" |]
-    // Indexed access using dot
-    let first = array1.[0]
+    // Create
+    let array1 = [| "a"; "b"; "c" |]
+    let array2 =
+        [| 1
+           2 |]
+    let array3 = [| 1..2..9 |]  // start..skip..last; [| 1; 3; 5; 7; 9 |]
 
-### Sequences
-A *sequence* is a logical series of elements of the same type. Individual sequence elements are computed only as required, so a sequence can provide better performance than a list in situations in which not all the elements are used.
+    // Indexed access
+    let first = array1[0]  // "a"
 
-    // Sequences can use yield and contain subsequences
-    let seq1 =
-        seq {
-            // "yield" adds one element
-            yield 1
-            yield 2
+    // Slicing is inclusive; [| "a"; "b" |]
+    let firstTwo = array1[0..1]
 
-            // "yield!" adds a whole subsequence
-            yield! [5..10]
-        }
+    // Assignment using `<-`
+    array1[1] <- "d"  // [| "a"; "d"; "c" |]
 
-### Higher-order functions on collections
-The same list `[ 1; 3; 5; 7; 9 ]` or array `[| 1; 3; 5; 7; 9 |]` can be generated in various ways.
+    // Pattern matching
+    match myArray with
+    | [| |] -> ...       // match an empty array
+    | [| 3 |] -> ...     // match array with single 3 item
+    | [| _; 4 |] -> ...  // match array with 2 items, second item = 4
 
- - Using range operator `..`
+See the [Array Module](https://fsharp.github.io/fsharp-core-docs/reference/fsharp-collections-arraymodule.html) for built-in functions.
 
-        let xs = [ 1..2..9 ]
+<div id="collections-sequences"></div>
 
- - Using list or array comprehensions
+### Sequences
 
-        let ys = [| for i in 0..4 -> 2 * i + 1 |]
+A *sequence* is a logical series of elements of the same type. Individual sequence elements are computed only as required,
```

Comment:
need to mention that seq is alias for `IEnumerable<T>` (which will be easier to explain if you also remove usage of `seq` as a suffix)